fileio/private/read_brainvision_vhdr.m

```diff
@@ -128,83 +128,88 @@
 hdr.impedances.ground=NaN;
 hdr.impedances.refChan=[];
 
-fid = fopen(filename, 'rt');
-if fid~=-1
-    while ~feof(fid)
-        tline = fgetl(fid);
-        if (length(tline) >= 9) && strcmp(tline(1:9),'Impedance')
-            chanCounter=0;
-            refCounter=0;
-            impCounter=0;
-            while chanCounter<hdr.NumberOfChannels && ~feof(fid)
-                chan_info = fgetl(fid);
-                if ~isempty(chan_info)
-                    impCounter=impCounter+1;
-                    [chanName,impedances] = strtok(chan_info,':');
-                    spaceList=strfind(chanName,' ');
-                    if ~isempty(spaceList)
-                        chanName=chanName(spaceList(end)+1:end);
-                    end;
-                    if strfind(chanName,'REF_')==1 %for situation where there is more than one reference
-                        refCounter=refCounter+1;
-                        hdr.impedances.refChan(refCounter)=impCounter;
-                        if ~isempty(impedances)
-                            hdr.impedances.reference(refCounter) = str2double(impedances(2:end));
-                        else
-                            hdr.impedances.reference(refCounter) = NaN;
-                        end
-                    elseif strcmpi(chanName,'ref') %single reference
-                        refCounter=refCounter+1;
-                        hdr.impedances.refChan(refCounter)=impCounter;
-                        if ~isempty(impedances)
-                            hdr.impedances.reference(refCounter) = str2double(impedances(2:end));
-                        else
-                            hdr.impedances.reference(refCounter) = NaN;
-                        end
-                    else
-                        chanCounter=chanCounter+1;
-                        if ~isempty(impedances)
-                            hdr.impedances.channels(chanCounter,1) = str2double(impedances(2:end));
-                        else
-                            hdr.impedances.channels(chanCounter,1) = NaN;
-                        end
-                    end;
-                end;
-            end
-            if ~feof(fid)
-                tline='';
-                while ~feof(fid) && isempty(tline)
-                    tline = fgetl(fid);
+try
+    fid = fopen_or_error(filename, 'rt');
+catch err
+    % quash
+    % TODO: Are we sure we want this to just silently return, instead of raising
```

Comment:
this should not have a try-catch, it should simply error inside fopen_or_error. Returning nothing does not make sense.